features/cryptocell/FEATURE_CRYPTOCELL310/sha1_alt.c

```diff
@@ -79,8 +91,21 @@ int mbedtls_sha1_finish_ret( mbedtls_sha1_context *ctx,
 int mbedtls_internal_sha1_process( mbedtls_sha1_context *ctx,
                                    const unsigned char data[64] )
 {
+    int ret = 0;
+    if( ctx->is_cc_initiated == 0 )
+    {
+        ret = init_cc( ctx );
```

Comment:
I don't think that calling `mbedtls_internal_sha1_process()` should be supported for several reasons:
1. This is an internal function, users shouldn't call it anyway.
2. If anybody, Mbed TLS, the tests or our users call it without calling `mbedtls_sha1_starts()` first, then they are violating the API

Reply:
As you can see::
[`sha1_process_wrap`](https://github.com/ARMmbed/mbedtls/blob/development/library/md_wrap.c#L366) calls `mbedtls_internal_sha1_process()`. I guess it's ok, since it's internal. The test for [`mbedtlos_md_process()`](https://github.com/ARMmbed/mbedtls/blob/development/tests/suites/test_suite_md.function#L11) only calls setup and process, so perhaps it's a bug in the test...

Reply:
I think it is a bug in the test, it should call `mbedtls_md_starts( )` too.

Reply:
created https://github.com/ARMmbed/mbedtls/issues/2227 to track this